ql/src/java/org/apache/hadoop/hive/ql/exec/vector/mapjoin/fast/VectorMapJoinFastHashTableLoader.java

```diff
@@ -314,12 +314,10 @@ public void load(MapJoinTableContainer[] mapJoinTables,
       } catch (InterruptedException e) {
         Thread.currentThread().interrupt();
         throw new HiveException(e);
-      } catch (IOException e) {
-        throw new HiveException(e);
-      } catch (SerDeException e) {
-        throw new HiveException(e);
       } catch (Exception e) {
         throw new HiveException(e);
+      } finally {
+        loadExecService.shutdownNow();
```

Comment:
I see that the `loadExecService` is cleanly handled (we may arrive here without creating a new one). This might be better:
```
        if (!loadExecService.isTerminated()) {
          loadExecService.shutdownNow();
        }
```
What do you think?

Reply:
Yes, Agree. Thanks for pointing this. Also added a null check.